packages/@uppy/status-bar/src/StatusBar.jsx

```diff
@@ -8,25 +8,56 @@ import StatusBarUI from './StatusBarUI.jsx'
 import packageJson from '../package.json'
 import locale from './locale.js'
 
-function getTotalSpeed (files) {
-  let totalSpeed = 0
-  files.forEach((file) => {
-    totalSpeed += getSpeed(file.progress)
-  })
-  return totalSpeed
+const speedFilterHalfLife = 2
+const ETAFilterHalfLife = 2
+
+let previousTotalBytes = 0
+function getCurrentSpeed (files, dt) {
+  const totalBytes = files.reduce((total, file) => {
+    return total + file.progress.bytesUploaded
+  }, 0)
+  if (totalBytes === 0) {
+    return 0
+  }
+  const newlyUploadedBytes = totalBytes - previousTotalBytes
+  previousTotalBytes = totalBytes
+  return newlyUploadedBytes / dt
 }
 
-function getTotalETA (files) {
-  const totalSpeed = getTotalSpeed(files)
-  if (totalSpeed === 0) {
+let previousSpeed = null
+function getInstantETA (files, dt) {
+  const currentSpeed = getCurrentSpeed(files, dt)
+  if (currentSpeed === 0) {
     return 0
   }
 
+  let filteredSpeed = currentSpeed
+  if (previousSpeed !== null) {
+    filteredSpeed = emaFilter(currentSpeed, previousSpeed, speedFilterHalfLife, dt)
+  }
+  previousSpeed = currentSpeed
+
   const totalBytesRemaining = files.reduce((total, file) => {
     return total + getBytesRemaining(file.progress)
   }, 0)
+  return (totalBytesRemaining / filteredSpeed)
+}
+
+let previousFilteredValue = null
+let lastUpdateTime = null
+function getFilteredETA (files) {
+  const time = Date.now() / 1000
```

Comment:
We might prefer to use `performance.now()` here instead incase the user changes their system clock in the middle on an upload – albeit that might be a bit far-fetched use case.